alonzo/impl/src/Cardano/Ledger/Alonzo/PlutusScriptApi.hs

```diff
@@ -242,6 +247,8 @@ evalScripts tx ((AlonzoScript.PlutusScript pscript, ds, units, cost) : rest) =
 --     3) The script is a twoPhase script, and the _txrdmrs Map of the TxWitness,
 --        contains Data for the script.
 checkScriptData ::
+-- ^^ This looks like it is also meant to check that the tx is carrying the right datum for a (nonnative) script
+-- this is why UTxO is needed (to get the datum hashes from the outputs)
```

Comment:
@polinavino - I forget what we decided to do about this one. Do you remember?

Reply:
I think it is fine (ie do nothing) - the datum check is separate